src/map/script.cpp

```diff
@@ -25686,6 +25722,221 @@ BUILDIN_FUNC( openstylist ){
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	struct item *itm = &sd->inventory.u.items_inventory[idx];
+
+	switch( script_getnum(st, 3) ) {
+	case SID_ID: {
+		std::shared_ptr<item_data> id;
+
+		if (script_isstring(st, 4))
+			id = item_db.searchname( script_getstr(st, 4) );
+		else
+			id = item_db.find( script_getnum(st, 4) );
+
+		if (id == nullptr) {
+			ShowError("buildin_setitemdata: Nonexistant item '%s'.\n", script_getstr(st, 4));
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		if (id->type != IT_ARMOR && id->type != IT_WEAPON && id->type != IT_SHADOWGEAR) {
+			ShowError("buildin_setitemdata: Invalid type for item '%s'. The item must be equippable.\n", script_getstr(st, 4));
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		itm->nameid = id->nameid;
+		break;
+	}
+	case SID_IDENTIFY:
+		itm->identify = script_getnum(st, 4) > 0;
+		break;
+	case SID_REFINE:
+		itm->refine = cap_value( script_getnum(st, 4), 0, MAX_REFINE );
+		break;
+	case SID_ATTRIBUTE:
+		itm->attribute = script_getnum(st, 4) > 0;
+		break;
+	case SID_CARD1:
+		itm->card[0] = script_getnum(st, 4);
+		break;
+	case SID_CARD2:
+		itm->card[1] = script_getnum(st, 4);
+		break;
+	case SID_CARD3:
+		itm->card[2] = script_getnum(st, 4);
+		break;
+	case SID_CARD4:
+		itm->card[3] = script_getnum(st, 4);
+		break;
+	case SID_EXPIRE:
+		itm->expire_time = (unsigned int)(time(NULL) + script_getnum(st, 4));
+		break;
+	case SID_BOUND: {
+		int bound = script_getnum(st, 4);
+		if (bound < BOUND_NONE || bound >= BOUND_MAX)
+			itm->bound = BOUND_NONE;
+		else
+			itm->bound = static_cast<bound_type>( bound );
+		break;
+	}
+	case SID_GRADE:
+		itm->enchantgrade = cap_value( script_getnum(st, 4), 0, 3 );
+		break;
+	case SID_OPTIONID1:
+		itm->option[0].id = script_getnum(st, 4);
+		break;
+	case SID_OPTIONID2:
+		itm->option[1].id = script_getnum(st, 4);
+		break;
+	case SID_OPTIONID3:
+		itm->option[2].id = script_getnum(st, 4);
+		break;
+	case SID_OPTIONID4:
+		itm->option[3].id = script_getnum(st, 4);
+		break;
+	case SID_OPTIONID5:
+		itm->option[4].id = script_getnum(st, 4);
+		break;
+	case SID_OPTIONVALUE1:
+		itm->option[0].value = script_getnum(st, 4);
+		break;
+	case SID_OPTIONVALUE2:
+		itm->option[1].value = script_getnum(st, 4);
+		break;
+	case SID_OPTIONVALUE3:
+		itm->option[2].value = script_getnum(st, 4);
+		break;
+	case SID_OPTIONVALUE4:
+		itm->option[3].value = script_getnum(st, 4);
+		break;
+	case SID_OPTIONVALUE5:
+		itm->option[4].value = script_getnum(st, 4);
+		break;
+	case SID_OPTIONPARAM1:
+		itm->option[0].param = script_getnum(st, 4);
+		break;
+	case SID_OPTIONPARAM2:
+		itm->option[1].param = script_getnum(st, 4);
+		break;
+	case SID_OPTIONPARAM3:
+		itm->option[2].param = script_getnum(st, 4);
+		break;
+	case SID_OPTIONPARAM4:
+		itm->option[3].param = script_getnum(st, 4);
+		break;
+	case SID_OPTIONPARAM5:
+		itm->option[4].param = script_getnum(st, 4);
+		break;
+	}
+
+	if (itm->equip > 0)
```

Comment:
Maybe store equip flag here and equip the item again afterwards?

Reply:
I left it like that because it's more versatile. User may (or not) want to auto-reequip the item. `equip` script command can do the job
```
*equip <item id>{,<char_id>};
*autoequip <item id>,<option>;

These commands are to equip a equipment on the attached character.
The equip function will equip the item ID given when the player has
this item in his/her inventory, while the autoequip function will
equip the given item ID when this is looted. The option parameter of
the autoequip is 1 or 0, 1 to turn it on, and 0 to turn it off.
```